### PR TITLE
Allow info to be null on getProfileInfo

### PR DESCRIPTION
**Pull request by Half-Shot**
We use this in various places on the Slack bridge to get the full profile of a user.

### CODE REVIEW

File: src/components/intent.ts [hunk 1]
```diff
@@ -520,7 +520,7 @@ export class Intent {
      * @return A Promise that resolves with the requested user's profile
      * information
      */
-    public async getProfileInfo(userId: string, info: string, useCache=true) {
+    public async getProfileInfo(userId: string, info: "avatar_url"|"displayname"|null = null, useCache = true) {
```
Comment (jaller94) [inline src/components/intent.ts:523]: Are "avatar_url"|"displayname" the only values that `profile.get()` supports?
TypeScript becomes a pain if Types are incorrectly too strict.

But if they are the only reasonable values, great to help bridges using this library to spell them correctly.

**Review by jaller94 — APPROVED** [review]
Looks good to me, however, I'm concerned that the two string may be too specific.

**Review by jaller94 — CHANGES_REQUESTED** [review]
Ah. the builds are failing.

File: src/components/intent.ts (hunk 1 above)
Comment (Half-Shot) [reply src/components/intent.ts:523]: The spec doesn't allow arbitrary keys, and it's been that way for as long as Matrix has been around. We can update this if the spec changes at some point.

**Review by jaller94 — APPROVED** [review]
giphub

**Pull request merged**